pkg/binding/bindingpolicy-resolver.go

```diff
@@ -200,58 +201,55 @@ func (resolver *bindingPolicyResolver) NoteObject(bindingPolicyKey string,
 			bindingPolicyKey)
 	}
 
-	// noteObject is thread-safe
-	changed, err := bindingPolicyResolution.noteObject(obj)
-	if err != nil {
-		return false, fmt.Errorf("failed to update resolution for bindingpolicy %v: %w", bindingPolicyKey, err)
-	}
-
-	return changed, nil
+	// ensureObjectIdentifier is thread-safe
+	return bindingPolicyResolution.ensureObjectIdentifierWithVersion(objIdentifier, resourceVersion), nil
 }
 
-// RemoveObject removes the given object from the maintained bindingpolicy's
-// objects resolution for the given bindingpolicy key.
+// RemoveObjectIdentifier removes the given object identifier from the
+// resolution for the given bindingpolicy key.
 //
 // The returned bool indicates whether the bindingpolicy resolution was changed.
-func (resolver *bindingPolicyResolver) RemoveObject(bindingPolicyKey string,
-	obj runtime.Object) bool {
+// If no resolution is associated with the given key, false is returned.
+func (resolver *bindingPolicyResolver) RemoveObjectIdentifier(bindingPolicyKey string,
+	objIdentifier util.ObjectIdentifier) bool {
 	bindingPolicyResolution := resolver.getResolution(bindingPolicyKey) // thread-safe
 
 	if bindingPolicyResolution == nil {
 		return false
 	}
 
-	// removeObject is thread-safe
-	return bindingPolicyResolution.removeObject(obj)
+	// removeObjectIdentifier is thread-safe
+	return bindingPolicyResolution.removeObjectIdentifier(objIdentifier)
 }
 
-// SetDestinations updates the maintained bindingpolicy's
-// destinations resolution for the given bindingpolicy key.
-// The given destinations set is expected not to be mutated after this call.
-func (resolver *bindingPolicyResolver) SetDestinations(bindingPolicyKey string,
-	destinations sets.Set[string]) {
+// GetObjectIdentifiers returns a copy of the object identifiers associated
+// with the given bindingpolicy key.
+// If no resolution is associated with the given key, an error is returned.
```

Comment:
I think you are referring to the single case this is done, which is when checking if the resolver's `EnsureObjectIdentifierWithVersion` returned an error prefixed with `bindingPolicyResolutionNotFoundErrorPrefix`, during the call to controller::updateResolutions for a given object identifier.

In this case, `EnsureObjectIdentifierWithVersion` does have to return this error, but in the special scenario noted here:
```
resolutionUpdated, err := c.bindingPolicyResolver.EnsureObjectIdentifierWithVersion(bindingPolicy.GetName(),
			objIdentifier, objMR.GetResourceVersion())
		if err != nil {
			if errorIsBindingPolicyResolutionNotFound(err) {
				// this case can occur if a bindingpolicy resolution was deleted AFTER
				// starting this iteration and BEFORE getting to the NoteObject function,
				// which occurs if a bindingpolicy was deleted in this time-window.
				logger.V(4).Info("skipped EnsureObjectIdentifierWithVersion for object-identifier (%v) and "+
					"bindingpolicy %s because the latter was deleted", objIdentifier, bindingPolicy.GetName())
				continue
			}
		...
```

An error should not be reported. Do you think it is fine to keep this single instance of checking the error? Should the check be changed to a more common implementation? Or is this single-case usage fine?